service/matching/tasklist/task_completer.go

```diff
@@ -189,16 +189,25 @@ func isDecisionTaskStarted(task *InternalTask, workflowExecutionResponse *types.
 		return true
 	}
 
+	logger.Debug(fmt.Sprintf("Decision task not started. TaskID: %v, ScheduleID: %v, PendingDecisionScheduleID: %v, PendingDecisionState: %v",
+		task.Event.TaskID, task.Event.ScheduleID, workflowExecutionResponse.PendingDecision.ScheduleID, *workflowExecutionResponse.PendingDecision.State),
+		tag.WorkflowID(task.Event.WorkflowID), tag.WorkflowRunID(task.Event.RunID))
+
 	return false
 }
 
-func isActivityTaskStarted(task *InternalTask, workflowExecutionResponse *types.DescribeWorkflowExecutionResponse) bool {
+func isActivityTaskStarted(task *InternalTask, workflowExecutionResponse *types.DescribeWorkflowExecutionResponse, logger log.Logger) bool {
```

Comment:
I would consider swapping this to be a method rather than passing the log along if you need to do metrics/logging, ie making this `func (tc *taskCompleterImpl) isActivityTaskStarted` so that you can do things like `dc.logger.debug(...` and metrics